src/components/Login/Login.jsx

```diff
@@ -16,21 +16,46 @@ export class Login extends React.Component {
           value: event.target.value
         })
     }
+    checkSpecialChar(string) {        
+        var format = /[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/;
+        if (format.test(string)){
+            return true;
+        } else {
+            return false;
+        }
+    }
+    checkUsername() {
+        if (this.state.value === '') {
+            window.alert('Please provide a username');
```

Comment:
You might favour just not sending the response if there is an empty string. Could be a bit annoying for the user to have a window alert.